mesonbuild/compilers/mixins/visualstudio.py

```diff
@@ -180,7 +175,12 @@ def get_output_args(self, outputname: str) -> T.List[str]:
         return ['/Fo' + outputname]
 
     def get_debug_args(self, is_debug: bool) -> T.List[str]:
-        return msvc_debug_args[is_debug]
+        if is_debug:
+            if any('ccache' in x for x in self.exelist):
```

Comment:
Since [sccache also requires `/Z7`](https://github.com/mozilla/sccache/blob/36c8b89337f28ce2efbd63f328b06ac1f0ff1d56/README.md?plain=1#L139-L160) I guess it would make sense to check for it as well.